processor/resourcedetectionprocessor/README.md

```diff
@@ -17,6 +17,16 @@ details of which are currently pending confirmation in the OpenTelemetry specifi
     * host.name
     * os.type
 
+Use the Docker detector (see below) if running the Collector as a Docker container.
+
+* Docker metadata: Queries the Docker daemon to retrieve the following resource attributes from the host machine:
+
+    * host.name
```

Comment:
Can you describe the difference between retrieving these with normal means? I'm surprised we need to use the docker API to get simple looking attributes.

Maybe it's to retrieve the information about the host, not the docker container. If so, this seems wrong, I believe our semantic conventions intend these to be filled for the current process, which would be using the values from within the docker container, not the host machine. I'd expect us to define new conventions in the spec, something like `containerhost.host.name`, `containerhost.os.type` for things outside the process.

Sorry if I'm missing something

Reply:
Also I was expecting this PR to implement `container` attributes

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/container.md#container

Surprised to see `host.name` and `os.type` in the docker detector :)

Reply:
> Can you describe the difference between retrieving these with normal means? I'm surprised we need to use the docker API to get simple looking attributes.

Here is the output of the `hostname` command and the `Name` field from the Docker API info endpoint from an Ubuntu Docker image running on an Ubuntu VM:

```
❯ docker run --rm -v /var/run/docker.sock:/var/run/docker.sock -it ubuntu bash
root@3d639eae3af3:/# apt-get update && apt-get install -y curl jq
# ...
root@3d639eae3af3:/# hostname
3d639eae3af3
root@3d639eae3af3:/# curl --silent --unix-socket /var/run/docker.sock http://localhost/info | jq .Name
"ubuntu-vm"
```

`hostname` gives you the `os.Hostname` (which is a random string on Docker). The Docker API gives you the name of the host machine (on Linux and Windows).

Similarly, the `os.type` will give you the host OS type (so, e.g., `windows` even if your container is Linux-based).

> I'd expect us to define new conventions in the spec, something like containerhost.host.name, containerhost.os.type for things outside the process.

I can definitely do that, but I fail to see why someone would care about the Docker hostname: the whole point (at least to me) of running the Collector as a Docker image is to not have other stuff running in the same Docker image.

> Also I was expecting this PR to implement container attributes

👍 will do

Reply:
Ah, thanks for the explanation, that's the opposite behavior of what I was expecting for `os.type`, surprised native calls within the container don't return information about the container :O And populating something other than the random string seems good.

Just curious, do you know how common it is to have the docker API exposed? For example, will kubernetes users generally be able to use the detector?

Reply:
> Just curious, do you know how common it is to have the docker API exposed? For example, will kubernetes users generally be able to use the detector?

I don't know about Kubernetes specifically, I would expect that people have to mount the Unix socket manually. We use this on our Datadog Agent for users running Docker by itself, but we have a different solution for Kubernetes.